### DISCUSSION TITLE
Überskalierung bei reinen PV Systemen zum Ausgleich von verschattungen

### DISCUSSION THREAD

**Discussion by spcqike** [hoylabs/OpenDTU-OnBattery#819] [Ideas]
Hallo zusammen 

ich möcht nochmal das Thema „überskalierung“ in den Raum werfen, da mir dies heute wieder aufgefallen ist und ich ein screen habe, der hoffentlich gut zeigt was ich meine :) 

<img width="375" alt="image" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/d2effca1-4025-4a05-84d3-512ebfc4b023">

ab ca.13:30 fällt die Leistung auf einem Eingang durch die wandernde Sonne, der zweite Eingang wird weiterhin rein durch das Limit begrenzt, könnte also die entstehende Lücke tendenziell ausgleichen. Die gelbe Linie (gesamt Leistung des WR) fällt flach, da nur auf einem Eingang. 

Erst ab ca 14:10 fällt die Gesamtleistung stärker, da nun beide Eingänge betroffen sind. 

Ich fänd es noch immer irgendwie ganz nett, wenn man im DPL bei reinen solar System sagen könnte „im Zweifel überskalieren, um verschattung eines oder mehrerer Eingänge auszugleichen“. 

Wir haben dies in Diskussionen und issues bereits mehrfach angesprochen, so eine Regelung hat natürlich ihre vor- aber auch Nachteile. Der größte Nachteil wäre wohl, dass man auch kurzfristig viel zu viel einspeisen könnte. (Bspw verschattung eines Paneels durch ne Wolke, das Limit wird vergrößert damit die verbleibenden mehr zur AC Leistung beitragen können und dann liefert auch das erste wieder volle Leistung.) 
Bei reinen PV Systemen ist dieses kurzfristige zu viel einspeisen (also verschenken) meiner Meinung nach aber weniger tragisch, als temporär Strom einzukaufen, den man auf den verbleibenden Paneelen hätte selbst produzieren können und es einfach nicht gemacht hat. 

Die Vorarbeit im DPL sollte nach den letzen Änderungen ja bereits fast vollständig drin sein. Die generelle Skalierung greift bereits nur noch bei wechselrichtern, wo jeder MPPT genau einen Eingang hat. Es bräuchte wohl nur noch einen Opt-In Schalter und die entsprechende Berechnung, die wir in den issues bereits irgendwo aufgeführt haben.

Schöner wäre natürlich, wenn hoymiles das in der Firmware direkt selbst regelt und ein AC Limit unabhängig von den Trackern geregelt wird, aber das ist aktuell halt einfach nicht der Fall. Und, sollte solch eine überskalierung ein Opt-In werden, kann man es bei neueren wechselrichter, so sie denn eine andere Firmware hätten, ja einfach deaktiviert lassen. 

Wie seht ihr das? 
Gibt es weitere rein PV Anwender die von solch einer Steuerung profitieren würden? 

Grüße

**Comment by hex2647** [comment]
bekommst ein like... aber soweit waren wir ja schon :)

**Reply by JaMo523** [reply]
Wenn man genug PV Leistung hat macht es eigentlich immer Sinn ein bisschen was ins Netz zu speisen. 

Mein Systemaufbau ist zwar anders aber ich habe bei mir die DPL Berechnung auch angepasst. Wenn der MPPT vom Victron mehr Leistung bereit stellt als benötigt wird, dann speise ich 50W ein.

**Reply by spcqike** [reply]
wie du siehst, speise ich dort ja ebenfalls 25W ein, dauerhaft. aber eben nur solange wie die Module gleichmäßig Licht bekommen. 

ja, man könnte es auch einfach komplett frei laufen lassen und den Überschuss verschenken (ich hab ja keinen Akku), abe das ist auch nicht immer sinnvoll. es gibt bereits viele und es wird noch mehr Situationen geben, in denen das nicht zwingend gut für das Netz ist. ne einzelne Anlage fällt da nicht auf, aber es werden ja auch immer mehr. 

gestern bspw. war wieder so ein Tag, an dem ein "verschenken" des Stroms kein "Geschenk" war, sondern zu Lasten anderer, am Ende zu Lasten der Allgemeinheit, verkauft wurde.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/00c6db74-270e-4ba3-bd64-6303634ffc58)

in diesen Fällen würde ich doch gern, wenn auch nur fürs Gewissen, nichts einspeisen, aber meinen Bedarf dennoch decken 😄

**Comment by AndreasBoehm** [comment]
Da dieses Feature auch für mich sehr praktisch wäre, setze mich die Tage dran und versuche es zu implementieren.

**Reply by hex2647** [reply]
Sehr cool… freue mich aufs testen

**Reply by spcqike** [reply]
Es gibt einen offenen PR
https://github.com/helgeerbe/OpenDTU-OnBattery/pull/956

Leider finde ich dort keine binary, so wie es bei @schlimmchen seinen PRs ist. 

Würde die Version, sofern sie auf einem 4Mb esp32 läuft, ebenfalls gern testen :)

**Reply by schlimmchen** [reply]
Das hat damit zu tun, dass Andreas ein "first-time contributor" ist und für solche der Workflow erst freigegeben werden muss. Ich hab seinen PR angeschaut und den Workflow gestartet.

~~@spcqike Ich lasse dieses Issue automatisch schließen, wenn #956 ge-merge't wird, ist das okay?~~ Vergiss es, das ist ja eine Diskussion hier, kein Issue :man_facepalming:
